<launch>
  <arg name="publish_pos" default="true"/>
  <arg name="publish_wrist" default="true"/>
  <arg name="left_arm" default="true"/>
  <arg name="right_arm" default="true"/>
  <arg name="shadow_hand" default="true"/>
  <arg name="left_gripper" default="true"/>

  <group if="$(arg right_arm)">
   <node name="pr2_right_arm_control"  pkg="unity_pr2_beginner" type="pr2_arms_control" output="screen">
        <param name="publish_pos" value="$(arg publish_pos)"/>
        <param name="publish_wrist" value="$(arg publish_wrist)" />
        <rosparam command="load" file="$(find unity_pr2_beginner)/config/pr2_right_arm_jog.yaml"/>
   </node>
  </group>

  <group if="$(arg left_arm)">
     <node name="pr2_left_arm_control"  pkg="unity_pr2_beginner" type="pr2_arms_control" output="screen">
        <param name="publish_pos" value="$(arg publish_pos)"/>
        <param name="publish_wrist" type="bool" value="false" />
        <rosparam command="load" file="$(find unity_pr2_beginner)/config/pr2_left_arm_jog.yaml"/>
      </node>

    </group>

    <group if="$(arg left_gripper)">
      <node name="pr2_left_gripper_control" pkg="unity_pr2_beginner" type="pr2_left_gripper_control" output="screen"/>
  </group>

  <group if="$(arg shadow_hand)">
    <!-- <node name="pr2_shadow_hand_contol"  pkg="unity_pr2_beginner" type="pr2_shadow_hand_contol" output="screen">
         <param name="publish_pos" value="$(arg publish_pos)"/>
         <param name="show_hand_inRviz" value="true"/>
         <rosparam command="load" file="$(find unity_pr2_beginner)/config/pr2_shadow_hand.yaml"/>
    </node> -->
    <node ns="hand" name="sr_parallel_grasp_controller" pkg="unity_pr2_beginner" type="sr_parallel_grasp_controller" output="screen" />
  </group>

</launch>
